storage/range.go

```diff
@@ -1252,6 +1259,8 @@ func (r *Range) InternalRangeLookup(batch engine.Engine, args *proto.InternalRan
 		reply.SetGoError(util.Errorf(
 			"Range lookup specified invalid maximum range count %d: must be > 0", rangeCount))
 		return
+	} else if args.IgnoreIntents {
+		rangeCount = 1 // simplify lookup because we may have to retry to read new
 	}
 
```

Comment:
would add something to the description of InternalRangeLookup that hints at the fact that there's a special consistency handling here and that ReadConsistency is ignored.